core/storage/base_model/gae_models.py

```diff
@@ -1619,8 +1626,8 @@ def get_snapshots_metadata(
                     'Invalid version number %s for model %s with id %s'
                     % (version_numbers[ind], cls.__name__, model_instance_id))
 
-        # We need the cast here to make sure that returned_models only contains
-        # BaseSnapshotMetadataModel and not None.
+        # Here we use cast because we need to make sure that returned_models
+        # only contains BaseSnapshotMetadataModel and not None.
```

Comment:
This one seems odd. If it actually ends up being None then what does casting it into a BaseSnapshotMetadataModel do?

Reply:
In the for loop just above it, we are already checking if these values are none or not. -- comment is also updated